legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/relationalMappingExecution.pure

```diff
@@ -118,6 +118,13 @@ function meta::relational::mapping::generateInstantiationExecutionNode(sq:meta::
    ]);
 }
 
+function meta::relational::mapping::compareTypesInRelationalPropertyMapping(r:RelationalPropertyMapping[0..1]):Boolean[0..1]
```

Comment:
call it assertCompatibleTypesInRelationalPropertyMapping, and use a assertion here. Assert will raise a PureAssertFailException on java side, which you can catch and rethrow